src/main/modules/document-download/index.ts

```diff
@@ -28,18 +19,36 @@ export class DocumentDownloadMiddleware {
 
     const dmStoreProxyForApplicationPdf = {
       endpoints: ['/downloads/divorce-application', '/downloads/application-to-end-civil-partnership'],
-      target: config.get('services.documentManagement.url'),
+      path: (req: AppRequest) => {
+        return req.session.userCase.documentsGenerated.find(doc => doc.value.documentType === DocumentType.APPLICATION)
+          ?.value.documentLink.document_binary_url;
+      },
     };
 
     const dmStoreProxyForRespondentAnswersPdf = {
       endpoints: ['/downloads/respondent-answers'],
-      target: config.get('services.documentManagement.url'),
+      path: (req: AppRequest) => {
+        return req.session.userCase.documentsGenerated.find(
+          doc => doc.value.documentType === DocumentType.RESPONDENT_ANSWERS
+        )?.value.documentLink.document_binary_url;
+      },
+    };
+
+    const dmStoreProxyForCertificateOfServicePdf = {
+      endpoints: ['/downloads/certificate-of-service'],
+      path: (req: AppRequest) => {
+        return req.session.userCase.alternativeServiceOutcomes.find(
+          doc =>
+            doc.value.successfulServedByBailiff === YesOrNo.YES &&
+            doc.value.certificateOfServiceDocument.documentType === DocumentType.CERTIFICATE_OF_SERVICE
```

Comment:
do you know that the CERTIFICATE_OF_SERVICE is the correct document? Because my ticket also stated it was the CERTIFICATE_OF_SERVICE but it was DEEMED_AS_SERVICE_GRANTED and DISPENSE_WITH_SERVICE_GRANTED

also since the deemed and dispensed are both also certificates of service it might be good to specify that that download link is specifically for bailiff certificate of service:
e.g '/bailiff-certificate-of-service'

Reply:
how do you confirm this?
i guess i can use the same find in my content and check if bailiff was successful. not sure about anything else as when i went through this service, it was CERTIFICATE_OF_SERVICE.